aquadoggo/src/graphql/client/mutation.rs

```diff
@@ -407,84 +414,176 @@ mod tests {
         #[case] expected_error_message: &str,
-        #[future]
-        #[from(test_db)]
-        db: TestSqlStore,
+        #[from(test_db)] runner: TestDatabaseRunner,
     ) {
-        let db = db.await;
+        let entry_encoded = entry_encoded.to_string();
+        let operation_encoded = operation_encoded.to_string();
+        let expected_error_message = expected_error_message.to_string();
 
-        let (tx, _rx) = broadcast::channel(16);
-        let context = HttpServiceContext::new(db.store, tx);
-        let client = TestClient::new(build_server(context));
+        runner.with_db_teardown(move |db: TestDatabase| async move {
+            let (tx, _rx) = broadcast::channel(16);
+            let context = HttpServiceContext::new(db.store, tx);
+            let client = TestClient::new(build_server(context));
 
-        let publish_entry_request = publish_entry_request(entry_encoded, operation_encoded);
+            let publish_entry_request = publish_entry_request(&entry_encoded, &operation_encoded);
 
-        let response = client
-            .post("/graphql")
-            .json(&json!({
-              "query": publish_entry_request.query,
-              "variables": publish_entry_request.variables
+            let response = client
+                .post("/graphql")
+                .json(&json!({
+                  "query": publish_entry_request.query,
+                  "variables": publish_entry_request.variables
+                }
+                ))
+                .send()
+                .await;
+
+            let response = response.json::<serde_json::Value>().await;
+            for error in response.get("errors").unwrap().as_array().unwrap() {
+                assert_eq!(
+                    error.get("message").unwrap().as_str().unwrap(),
+                    expected_error_message
+                )
             }
-            ))
-            .send()
-            .await;
+        });
+    }
 
-        let response = response.json::<serde_json::Value>().await;
-        for error in response.get("errors").unwrap().as_array().unwrap() {
-            assert_eq!(
-                error.get("message").unwrap().as_str().unwrap(),
-                expected_error_message
+    #[rstest]
+    fn publish_many_entries(
+        #[from(test_db)]
+        #[with(100, 1, true, TEST_SCHEMA_ID.parse().unwrap())]
+        runner: TestDatabaseRunner,
+    ) {
+        runner.with_db_teardown(|populated_db: TestDatabase| async move {
+            // Get the author.
+            let author = Author::try_from(
+                populated_db
+                    .key_pairs
+                    .first()
+                    .unwrap()
+                    .public_key()
+                    .to_owned(),
             )
-        }
+            .unwrap();
+
+            // Setup the server and client with a new empty store.
+            let (tx, _rx) = broadcast::channel(16);
+            let new_store = initialize_store().await;
```

Comment:
This test uses 2 stores which causes problems with multiple db connections. I imagine it can just be refactored somehow.